app/controllers/shops_controller.rb

```diff
@@ -60,14 +60,13 @@ def change_status
     @shop = Shop.find(params[:shop_id])
     authorize! :show, @shop
     @appointment = Appointment.find(params[:appointment])
-    case params[:status]
-    when 'ready'
+    if @appointment.booked
```

Comment:
don't you mean @appointment.booked?

also this is not what I asked. You need to test the action:  @appointment.ready_for_pickup! because it can fail. and have a good / bad flash message depending on outcome